LAVA_setup_guide.md

```diff
@@ -0,0 +1,172 @@
+The process below describes how to do it from scratch.
+
+### OS setup
+
+Lava uses debian buster as base for both server and workers.
+
+- Install debian buster https://www.debian.org/releases/buster/debian-installer/
+- Log in as root
+- `usermod -a -G sudo <username>`
+- `echo "deb https://apt.lavasoftware.org/release buster main" >> /etc/apt/sources.list`
+- you may now login as user
+- `wget https://apt.lavasoftware.org/lavasoftware.key.asc`
+- `sudo apt-key add lavasoftware.key.asc`
+- `sudo apt-get update`
+
+Note: although it is not officially supported, lava-server can be installed in Ubuntu (e.g. on WSL) by adding the below step:
+`sudo apt-get install postgresql && sudo service postgresql start`
+
+This resolves a dpkg-configure error that occurs because Postgres is not listening to port 5432 during installation
+
+### Server setup
+
+Here is the process you'll need to build the server from scratch. It's recommended you use the docker compose instead -
+see below. 
+
+#### Install LAVA server
+
+- `sudo apt install lava-server`
+- `sudo a2dissite 000-default`
+- `sudo a2enmod proxy`
+- `sudo a2enmod proxy_http`
+- `sudo a2ensite lava-server.conf`
+- `sudo service apache2 restart`
+
+#### Setup user
+
+- `sudo lava-server manage users add <usernane> --passwd <password>`
+- `sudo lava-server manage authorize_superuser --username <usernane>`
+
+#### Set website URL (this affect internal web interface links)
+
+- go to http://localhost/admin/sites/site/
+- click on example.com and delete it
+- add new one with the correct url
+- edit /usr/lib/python3/dist-packages/lava_server/settings/common.py
+    * search for SITE_ID and change it to SITE_ID = 2
+    * `sudo service lava-server-gunicorn restart`
+    * any time you change site domain you have to change the index to match
+
+#### Add Worker
+
+- `sudo lava-server manage workers add <WORKER_NAME>`
+    * don't do that on a single machine install, it will break lava
+    * set the hostname to the worker name
+- copy the token string to be used when configuring the worker
+
+#### Add device definition
+
+- create a new file in /etc/lava-server/dispatcher-config/devices with the name of the device (you can use the one from https://github.com/paul-szczepanek-arm/lava-prototype)
+    * You can just copy the whole directory https://github.com/paul-szczepanek-arm/lava-prototype/tree/master/lava-server to /etc/, otherwise follow steps below
+    * must have extension jinja2: `mbed.jinja2`
+    * the device should inherit from a device type in /usr/share/lava-server/device-types, to inherit from docker add line: `{% extends "docker.jinja2" %}`
+    * Must pass in extra parameters to docker: `{% set docker_extra_arguments = ["-v /dev:/dev --group-add=dialout --device-cgroup-rule 'a 166:* rwm'"] %}`
+- `sudo service lava-server-gunicorn restart`
+
+### Worker setup
+
+#### Install dependencies
+
+`sudo apt install docker.io git`
```

Comment:
This will install out dated packets. Wouldn't it be better to link to https://docs.docker.com/engine/install/debian/ ?

Reply:
OK